src/Integration/ErrorListenerIntegration.php

```diff
@@ -29,17 +29,21 @@ final class ErrorListenerIntegration implements IntegrationInterface
     /**
      * Constructor.
      *
-     * @param Options $options           The options to be used with this integration
-     * @param bool    $handleFatalErrors Whether to handle fatal errors or not
+     * @param Options|null $options           The options to be used with this integration
+     * @param bool         $handleFatalErrors Whether to handle fatal errors or not
      */
-    public function __construct(Options $options, bool $handleFatalErrors = true)
+    public function __construct(?Options $options = null, bool $handleFatalErrors = true)
```

Comment:
This is kind of an internal note to myself.
In JS for example we have specific options just for the integration that are not reflected in the global options.
In PHP we have all possible options in the global options object, so making the options deprecated here makes kinda sense. TBD if this should be homogenized between SDKs.

Reply:
I think that it makes sense to have per-integration options, definitely! Honestly I'm not sure about whether we should deprecate passing options to the constructor or just ignore them as this PR is basically fixing a bug. Anyway, I don't have in mind a way to move the existing options from the global configuration to each integration without breaking BC

Reply:
Maybe we just don't make it `optional` then the migration moving the options to the integration is a bit easier.

Reply:
We have to make it `optional` because otherwise we cannot throw the deprecation saying that passing global options is discouraged as we don't have a way to know if they did or not